kayenta-web/src/main/java/com/netflix/kayenta/interceptors/MetricsInterceptor.java

```diff
@@ -0,0 +1,142 @@
+ * <p>
+ * A `timer` will be created for each request with the following tags:
+ * <p>
+ * - controller name
+ * - controller method
+ * - status (2xx, 4xx, 5xx, etc.)
+ * - statusCode (200, 404, 500, etc.)
+ * - success (true/false depending on whether the request resulted in an exception)
+ * - cause (if success == false, the name of the raised exception)
+ */
+public class MetricsInterceptor extends HandlerInterceptorAdapter {
+  static final String TIMER_ATTRIBUTE = "Metrics_startTime";
+
+  private final Registry registry;
+  private final String metricName;
+  private final String contentLengthMetricName;
+  private final Set<String> queryParamsToTag = new HashSet<String>();
+  private final Set<String> controllersToExclude = new HashSet<String>();
+
+  /**
+   * @param registry             Underlying metrics registry
+   * @param metricName           Metric name
+   * @param queryParamsToTag     Variables from the request that should be added as metric tags
+   * @param controllersToExclude Controller names that should be excluded from metrics
+   */
+  public MetricsInterceptor(Registry registry,
+                            String metricName,
+                            Collection<String> queryParamsToTag,
+                            Collection<String> controllersToExclude
+  ) {
+    this.registry = registry;
+    this.metricName = metricName;
+    this.contentLengthMetricName = format("%s.contentLength", metricName);
+    if (queryParamsToTag != null) {
+      this.queryParamsToTag.addAll(queryParamsToTag);
+    }
+    if (controllersToExclude != null) {
+      this.controllersToExclude.addAll(controllersToExclude);
+    }
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+    request.setAttribute(TIMER_ATTRIBUTE, getNanoTime());
+    return true;
+  }
+
+  @Override
+  public void afterCompletion(HttpServletRequest request,
+                              HttpServletResponse response,
+                              Object handler,
+                              Exception ex) throws Exception {
+    if (handler instanceof HandlerMethod) {
+      HandlerMethod handlerMethod = (HandlerMethod) handler;
+
+      String controller = handlerMethod.getMethod().getDeclaringClass().getSimpleName();
+      if (controllersToExclude.contains(controller)) {
+        return;
+      }
+
+      Integer status = response.getStatus();
+      if (ex != null) {
+        // propagated exceptions should get tracked as '500' regardless of response status
+        status = 500;
+      }
+
+      Id id = registry.createId(metricName)
+        .withTag("controller", controller)
+        .withTag("method", handlerMethod.getMethod().getName())
+        .withTag("status", status.toString().charAt(0) + "xx")
+        .withTag("statusCode", status.toString());
+
+      List<String> paramNames = Collections.list(request.getParameterNames());
+
```

Comment:
Can you mark this as the place that differs, and perhaps todo generalizing kork in such a way as to support this use case so you dont need this variant?

Leave for now but note it.

Reply:
Done.